### ISSUE TITLE
False positive `C901` thrown for outer function in closure

### ISSUE THREAD

**Issue by jamesbraza** [charliermarsh/ruff#4384]
Please see this snippet:

```python
from functools import wraps


def wrapper(f):
    @wraps(f)
    def wrapped_f(*args, **kwargs):
        if 1 == 1:
            pass
        if 1 == 1:
            pass
        if 1 == 1:
            pass
        return f(*args, **kwargs)

    return wrapped_f
```

And the below `pyproject.toml`

```toml
[tool.ruff.mccabe]
max-complexity = 3

[tool.flake8]
max-complexity = 3
```

With `ruff==0.0.265` will print:

```bash
> ruff --select=C901 a.py
a.py:4:5: C901 `wrapper` is too complex (5 > 3)
a.py:6:9: C901 `wrapped_f` is too complex (4 > 3)
Found 2 errors.
> flake8 a.py
a.py:4:1: C901 'wrapper' is too complex (5)
```

The 2nd error about `wrapped_f` if correct, as its cyclomatic complexity is above 3.  However, I think the 1st error about the `wrapper` (outer) function is a false positive, as it only has a cyclomatic complexity of 1.

In other words, I think `ruff` has a false positive `C901` for closures, only the inner should throw.

For what it's worth, `flake8==6.0.0` does not have this false positive, but I think they mark the extra complexity in the wrong place (at `wrapper`, not `wrapped_f`).

**Comment by charliermarsh** [comment]
I think I agree with you, though I'd love to get other opinions since it would be a deviation from Flake8 (or rather, McCabe).

**Comment by travis-leith** [comment]
> I think I agree with you, though I'd love to get other opinions since it would be a deviation from Flake8 (or rather, McCabe).

@charliermarsh have a look at https://github.com/astral-sh/ruff/issues/19228. I think it is hard to argue that this function is complex. It should not be flagged imo.
